File: leetcode/1800-1899/1823. Find the Winner of the Circular Game/1823.cpp
class Solution
{
    int helper(int n, int k)
    {
        int res = 0;

        if (n == 1)
            return 1;

        res = (helper(n - 1, k) + k) % n;

        return res == 0 ? n : res;
    }

public:
    int findTheWinner(int n, int k) { return helper(n, k); }
};

// TC: O(N) SC:O(N)

File: leetcode/1800-1899/1823. Find the Winner of the Circular Game/1823.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:

        def helper(n, k):
            if n == 1:
                return 0 # 0 based index

            return (k + helper(n - 1, k)) % n

        return helper(n, k) + 1 # 0 based index but 1<=n
#TC:O(N)  #SC: O(n)